### PR TITLE
attempt at accepting an array as input

### PR DESCRIPTION
**Pull request by pagetribe**
Here is my attempt at allowing for an array to passed in.

### CODE REVIEW

File: test/locater_test.js [hunk 1]
```diff
@@ -56,8 +73,8 @@ describe("locater", function() {
     });
 
     it("matches multiple occurences of a regex in a line", function() {
-      var result = locater.find(/[a-zA-Z]{6}/g, 'kaskade - 4AM');
```
Comment (pagetribe) [inline test/locater_test.js:59]: Not sure if this test was behaving as expected as it was only matching once. [a-zA-Z]{6} -> kaskad

Let me know if I'm missing something with my change.

File: lib/locater.js [hunk 2]
```diff
@@ -7,52 +7,56 @@
...
-    sourceLines.forEach(function (line, idx) {
-      // Do not count the last line that is empty
-      if (idx === sourceLines.length - 1 && ! line) {
-        return;
-      }
-
-      var indices = findIndices(target, line);
-
-      indices.forEach(function (index) {
-        var position = {line: lineNumber, cursor: index + 1};
```
Comment (pagetribe) [inline lib/locater.js:43]: I thought we might be able to convert all instances of target to array so we can treat it the same for all input and run it through the forEach saving on if else statements.

File: lib/locater.js [hunk 3]
```diff
@@ -7,52 +7,56 @@
 function findIndices(target, line) {
   var indices = [];
 
-  if (typeof target === 'string') {
-    var index = 0;
-    var offset = 0;
```
Comment (pagetribe) [inline lib/locater.js:12]: Converting all calls here to regex to reduce code and remove if else statements.

**Comment by pagetribe** [comment]
First open source pull request let me know if the procedure was ok.

File: lib/locater.js [hunk 4]
```diff
@@ -7,52 +7,56 @@
...
 
-        results.push(position);
-      });
+    sourceLines.forEach(function (line, lineNumber) {
+      target.forEach(function(currentValue) {
+        var indices = findIndices(currentValue, line);
 
-      lineNumber++;
+        indices.forEach(function (index) {
+          results.push( { line: lineNumber + 1, cursor: index + 1 } );
```
Comment (sungwoncho) [inline lib/locater.js:50]: Why do we have `{ line: lineNumber + 1, cursor: index + 1 }` compared to `{ line: lineNumber, cursor: index + 1 }` in the old version?

_Edit_

nvm. Realized it is an index of an array. Maybe `lineNumber` is not a good name.

File: test/locater_test.js (hunk 1 above)
Comment (sungwoncho) [inline test/locater_test.js:59]: I think `find` here was meant to match twice: `kaskad` and `askade`. This was passing in `master`.

File: lib/locater.js [hunk 5]
```diff
@@ -7,52 +7,56 @@
...
+  while ((match =  target.exec(line)) !== null) {
+    indices.push(match.index);
   }
 
   return indices;
 }
 
+/**
+ * Removes the last line break from a string.
+ * @param line {String} - the string to remove line break from
```
Comment (sungwoncho) [inline lib/locater.js:24]: nbd, but I'd change `line` to `source`

File: lib/locater.js [hunk 6]
```diff
@@ -7,52 +7,56 @@
...
+ */
+function removeLastLineBreaks(source) {
+  return source.replace(/[\n]+$/, '')
+}
+
+/**
+ * Converts sting to lines based on \n. Also removes last line break
+ * @param line {String} - the string split
```
Comment (sungwoncho) [inline lib/locater.js:32]: same as above

**Comment by sungwoncho** [comment]
Merged. Fixed the test [here](https://github.com/sungwoncho/locater/commit/a596949467d430df08ffc167869e6759bbda8e4e). Published as v1.1.0. Thanks for the PR.

**Pull request merged**